classes/class-paysoncheckout-for-woocommerce-subscriptions.php

```diff
@@ -36,9 +36,10 @@ public function trigger_scheduled_payment( $renewal_total, $renewal_order ) {
 		$subscription_id = get_post_meta( $order_id, '_payson_subscription_id', true );
 
 		if ( empty( $subscription_id ) ) {
-			$subscription_id = get_post_meta( WC_Subscriptions_Renewal_Order::get_parent_order_id( $order_id ), '_payson_subscription_id', true );
-			$order->update_meta_data('_payson_subscription_id', $subscription_id );
-			$order->save();
+			$subscription = wc_get_order(WC_Subscriptions_Renewal_Order::get_parent_order_id( $order_id ));
+			$subscription_id = $subscription->get_meta('_payson_subscription_id')
+			$subscription->update_meta_data('_payson_subscription_id', $subscription_id );
```

Comment:
This is almost correct. The id is retrieved from the subscription, as it should be, but it should be saved to the order, not the subscription.

```php
$subscription_id = $subscription->get_meta('_payson_subscription_id')
$order->update_meta_data('_payson_subscription_id', $subscription_id );
$order->save();
```